packages/cli/test/unit/commands/promote.test.ts

```diff
@@ -0,0 +1,380 @@
+    const { cwd, previousDeployment } = initPromoteTest();
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.id}" in ${previousDeployment.creator?.username}`
+    );
+    await expect(client.stderr).toOutput('Promote in progress');
+    await expect(client.stderr).toOutput(
+      `Success! ${chalk.bold('vercel-promote')} was promoted to ${
+        previousDeployment.url
+      } (${previousDeployment.id})`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  it('should promote by deployment url', async () => {
+    const { cwd, previousDeployment } = initPromoteTest();
+    client.setArgv('promote', previousDeployment.url, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.url}" in ${previousDeployment.creator?.username}`
+    );
+    await expect(client.stderr).toOutput('Promote in progress');
+    await expect(client.stderr).toOutput(
+      `Success! ${chalk.bold('vercel-promote')} was promoted to ${
+        previousDeployment.url
+      } (${previousDeployment.id})`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  it('should get status while promoting', async () => {
+    const { cwd, previousDeployment, project } = initPromoteTest({
+      promotePollCount: 10,
+    });
+
+    // start the promote
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    promote(client);
+
+    // need to wait for the promote request to be accepted
+    await sleep(500);
+
+    // get the status
+    client.setArgv('promote', '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Checking promotion status of ${project.name}`
+    );
+    await expect(client.stderr).toOutput(
+      `Success! ${chalk.bold('vercel-promote')} was promoted to ${
+        previousDeployment.url
+      } (${previousDeployment.id})`
+    );
+
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  it('should error if promote request fails', async () => {
+    const { cwd, previousDeployment } = initPromoteTest({
+      promotePollCount: 10,
+      promoteStatusCode: 500,
+    });
+
+    client.setArgv('promote', previousDeployment.id, '--yes', '--cwd', cwd);
+    const exitCodePromise = promote(client);
+
+    await expect(client.stderr).toOutput(
+      `Fetching deployment "${previousDeployment.id}" in ${previousDeployment.creator?.username}`
+    );
+
+    // we need to wait a super long time because fetch will return on 500
+    await expect(client.stderr).toOutput('Response Error (500)', 20000);
+
+    await expect(exitCodePromise).resolves.toEqual(1);
+  });
+
+  it('should error if rollback fails (no aliases)', async () => {
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** 

```suggestion
  it('should error if promote fails (no aliases)', async () => {
```